integration-tests/modules/__tests__/link-modules/define-link.spec.ts

```diff
@@ -0,0 +1,92 @@
+import { medusaIntegrationTestRunner } from "medusa-test-utils"
+import CurrencyModule from "@medusajs/currency"
+import RegionModule from "@medusajs/region"
+import { defineLink } from "@medusajs/utils"
+import { MedusaModule } from "@medusajs/modules-sdk"
+
+jest.setTimeout(50000)
+
+medusaIntegrationTestRunner({
+  testSuite: ({ getContainer }) => {
+    describe("defineLink", () => {
+      let appContainer
+      let remoteQuery
+
+      beforeAll(async () => {
+        appContainer = getContainer()
+        remoteQuery = appContainer.resolve("remoteQuery")
+      })
+
+      it("should generate a proper link definition", async () => {
+        const currencyLinks = CurrencyModule.links
+        const regionLinks = RegionModule.links as any
+
+        const link = defineLink(currencyLinks.currency, regionLinks.region)
+
+        const linkDefinition = MedusaModule.getCustomLinks()
+          .map((linkDefinition: any) => {
+            const definition = linkDefinition(MedusaModule.getLoadedModules())
+            return definition.serviceName === link.serviceName && definition
+          })
+          .filter(Boolean)[0]
+
+        expect(link.serviceName).toEqual("currencyCurrencyRegionRegionLink")
+        expect(linkDefinition).toEqual({
```

Comment:
thought: `database` could probably also be autogenerated no?
```
databaseConfig: {
    tableName: [left_source_right_source],
    idPrefix: [first three characters of each source],
  },
```
For example:
databaseConfig: {
    tableName: "order_payment_collection",
    idPrefix: "ordpay",
  },

Reply:
everything is autogenerated here, what do you mean? the link module is already providing a default config for it, it is not specified because we do not need to override it here

Reply:
Ah sorry, if we are already autogenerating this, forget what I said 😄